packages/wallets/src/features/cashier/modules/Transactions/components/TransactionsCompleted/__tests__/TransactionsCompleted.spec.tsx

```diff
@@ -0,0 +1,181 @@
+        (useActiveWalletAccount as jest.Mock).mockReturnValue({ data: {}, isLoading: false });
+        (useAllAccountsList as jest.Mock).mockReturnValue({ data: [], isLoading: false });
+        (useInfiniteTransactions as jest.Mock).mockReturnValue({ data: null, isLoading: false, setFilter: jest.fn() });
+
+        render(
+            <CashierScrollContext.Provider value={{ onCashierScroll, setOnCashierScroll }}>
+                <TransactionsCompleted />
+            </CashierScrollContext.Provider>
+        );
+        expect(screen.getByText('No Data')).toBeInTheDocument();
+    });
+
+    it('should render transactions and group them by date', async () => {
+        (useActiveWalletAccount as jest.Mock).mockReturnValue({ data: {}, isLoading: false });
+        (useAllAccountsList as jest.Mock).mockReturnValue({ data: [], isLoading: false });
+        (useInfiniteTransactions as jest.Mock).mockReturnValue({
+            data: [
+                { transaction_time: 1643798400 }, // Date: 02 Feb 2022
+                { transaction_time: 1643712000 }, // Date: 01 Feb 2022
+            ],
+            isLoading: false,
+            setFilter: jest.fn(),
+        });
+
+        render(
+            <CashierScrollContext.Provider value={{ onCashierScroll, setOnCashierScroll }}>
+                <TransactionsCompleted />
+            </CashierScrollContext.Provider>
+        );
+        expect(screen.getByText('01 Feb 2022')).toBeInTheDocument();
+        expect(screen.getByText('02 Feb 2022')).toBeInTheDocument();
+        expect(screen.getAllByText('Transaction Row')).toHaveLength(2);
+    });
+
+    it('should fetch more transactions when bottom is reached', async () => {
+        const mockFetchNextPage = jest.fn();
+
+        (useActiveWalletAccount as jest.Mock).mockReturnValue({ data: {}, isLoading: false });
+        (useAllAccountsList as jest.Mock).mockReturnValue({ data: [], isLoading: false });
+        (useInfiniteTransactions as jest.Mock).mockReturnValue({
+            data: [{ transaction_time: 1643798400 }],
+            fetchNextPage: mockFetchNextPage,
+            isFetching: false,
+            isLoading: false,
+            setFilter: jest.fn(),
+        });
+
+        render(
+            <CashierScrollContext.Provider value={{ onCashierScroll, setOnCashierScroll }}>
+                <div
+                    data-testid='dt_transactions_completed'
+                    onScroll={onCashierScroll}
+                    style={{
+                        height: '300px',
+                        width: '1920px',
+                    }}
+                >
+                    <TransactionsCompleted />
+                </div>
+            </CashierScrollContext.Provider>
+        );
+
+        const el = screen.getByTestId('dt_transactions_completed');
+
+        act(() => {
+            fireEvent.scroll(el, { target: { scrollY: 100 } });
+        });
+
+        expect(mockFetchNextPage).toBeCalled();
```

Comment:
ideally - we should test the outcome, not mechanics, 

so test that new transactions actually got loaded and rendered, not just the fact that some function got called (I can call this function without rendering transactions)

Reply:
CMIIW but I believe this is not viable to emulate as the fetching function is used in connection with setOnCashierScroll from the cashier context provider. 

The scroll function itself happens in the grandparent component of TransactionsCompleted, which is the WalletCashier component, containing this context provider. The actual scroll function itself couldn't be fully emulated in this unit test for the grandchild of the component withholding the scroll context provider. The best solution is therefore to emulate that the scroll state has been passed to the provider and therefore calls the fetching function.